### PR TITLE
[Tests] NFC: Add a couple of tests for parametrized opaque result types

### PR DESCRIPTION
**Pull request by xedin**

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

File: test/type/opaque.swift [hunk 1]
```diff
@@ -590,3 +590,24 @@ func f62787_1(x: Bool) -> Optional<some Collection<Int>> {
...
+}
+
+do {
+  struct G<A: P1>: P3 {}
+
+  struct S: P1 {}
+
+  func test1() -> some P3<Int> { // expected-note {{opaque return type declared here}}
+    return G<S>()
+    // expected-error@-1 {{type of local function 'test1()' requires that 'S' conform to 'Int'}}
```
Comment (slavapestov) [inline test/type/opaque.swift:606]: do you want to fix the diagnostic to take requirement kind into account too? :)  It seems it was hardcoded to `RequirementKind::Conformance`

**Review by slavapestov — APPROVED** [review]

File: test/type/opaque.swift (hunk 1 above)
Comment (xedin) [reply test/type/opaque.swift:606]: Yeah, I'll look into that!

**Comment by xedin** [comment]
I'll follow-up with a same-type diagnostic improvement today.

**Pull request merged**